### ISSUE TITLE
Testing has a bug.

### ISSUE THREAD

**Issue by happsky** [kregmi/cross-view-image-synthesis#11]
`DATA_ROOT=./datasets/AB_AsBs name=a2g_fork which_direction=a2g phase=test which_epoch=35 th test_fork.lua`  
{
  input_nc : 3
  results_dir : "./results/"
  name : "a2g_fork"
  batchSize : 10
  phase : "test"
  fineSize : 256
  aspect_ratio : 1
  how_many : "all"
  gpu : 1
  nThreads : 1
  DATA_ROOT : "./datasets/AB_AsBs"
  serial_batch_iter : 1
  output_nc_seg : 3
  which_epoch : 35
  loadSize : 256
  cudnn : 1
  serial_batches : 1
  which_direction : "a2g"
  display : 0
  output_nc : 3
  preprocess : "regular"
  checkpoints_dir : "./checkpoints"
  display_id : 200
  flip : 0
}
Random Seed: 4974	
#threads...1	
Starting donkey with id: 1 seed: 4975
table: 0x41bfb1c8
./datasets/AB_AsBs
trainCache	/home/csdept/projects/cross-view-image-synthesis/cache/_home_csdept_projects_cross-view-image-synthesis_datasets_AB_AsBs_test_trainCache.t7
Creating train metadata
serial batch:, 	1
table: 0x40732bf0
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
now combine all the files to a single large file
load the large concatenated list of sample paths to self.imagePath
cmd..wc -L '/tmp/lua_jMFhFH' |cut -f1 -d' '
8 samples found........................... 0/8 .........................................]  ETA: 0ms | Step: 0ms         
Updating classList and imageClass appropriately
 [======================================== 1/1 ========================================>]  Tot: 0ms | Step: 0ms         
Cleaning up temporary files
Dataset Size: 	8	
checkpoints_dir	./checkpoints	
nn.gModule

No outputs generated.

**Comment by kregmi** [comment]
Dataset size(:8) being smaller than the batchSize(=10) caused the problem. You can try with smaller batchSize like 2 or 4 for this sample dataset.

**Comment by happsky** [comment]
Thanks for your quick response and it solves my problem.
I have another question that do you also use segmentation masks during the testing stage?

**Comment by nidetaoge** [comment]
Actually through the GNet it doesn't need to generate or use segmentation masks, but when you use the scripts in data/ to generate the inputs, it shares the same interface with the training code.

**Comment by kregmi** [comment]
@happsky We can generate the segmentation masks too but primary task is image generation.
@nidetaoge Yes, you are right.

**Comment by thanif** [comment]
DATA_ROOT=./datasets/AB_AsBs name=a2g_fork which_direction=a2g phase=test which_epoch=35 th test_fork.lua
{
  input_nc : 3
  results_dir : "./results/"
  name : "a2g_fork"
  batchSize : 2
  phase : "test"
  fineSize : 256
  aspect_ratio : 1
  how_many : "all"
  gpu : 1
  nThreads : 1
  DATA_ROOT : "./datasets/AB_AsBs"
  serial_batch_iter : 1
  output_nc_seg : 3
  which_epoch : 35
  loadSize : 256
  cudnn : 1
  serial_batches : 1
  which_direction : "a2g"
  display : 0
  output_nc : 3
  preprocess : "regular"
  checkpoints_dir : "./checkpoints"
  display_id : 200
  flip : 0
}
Random Seed: 3426	
#threads...1	
Starting donkey with id: 1 seed: 3427
table: 0x4059f688
./datasets/AB_AsBs
trainCache	/home/talha/cross-view-image-synthesis/cache/_home_talha_cross-view-image-synthesis_datasets_AB_AsBs_test_trainCache.t7
Creating train metadata
serial batch:, 	1
table: 0x41598ce8
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
now combine all the files to a single large file
load the large concatenated list of sample paths to self.imagePath
cmd..wc -L '/tmp/lua_R9aqen' |cut -f1 -d' '
8 samples found...... 0/8 ......................] ETA: 0ms | Step: 0ms          
Updating classList and imageClass appropriately
 [=================== 1/1 =====================>] Tot: 0ms | Step: 0ms          
Cleaning up temporary files
Dataset Size: 	8	
checkpoints_dir	./checkpoints	
nn.gModule
processing batch 1	
filepaths_curr: 	{
  1 : "4AyrBfjG-6ZbAa_uD57LcQ.x873.y450.a-170.jpg"
  2 : "_1h8nT4xM5i4cbsRubG8Tg.x1379.y503.a-148.jpg"
}
/home/talha/torch/install/bin/luajit: bad argument #2 to '?' (sizes do not match at /home/talha/torch/extra/cutorch/lib/THC/THCTensorCopy.cu:31)
stack traceback:
	[C]: at 0x7f51bbcf13e0
	[C]: in function '__newindex'
	/home/talha/cross-view-image-synthesis/util/util.lua:164: in function 'deprocess_batch'
	test_fork.lua:126: in main chunk
	[C]: in function 'dofile'
	...alha/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00405d50

@kregmi ,Can you please guide me about what the size used for training was ?
Alos @happsky, were you able to test the model using pre trained weights on the provided 8 images?

**Comment by thanif** [comment]
@happsky  and @kregmi, Sorry for the comment as I was using seg weights while fork code. My bad.
